src/clincoded/static/components/provisional_classification/snapshots.js

```diff
@@ -218,6 +232,13 @@ class CurationSnapshots extends Component {
             resourceParent = snapshot.resourceParent.interpretation;
         }
 
+        // Criteria to render a publish link (above a "View Approved Summary" button):
+        // User has permission to publish (this.props.allowPublishButton) and a publish event is not currently in progress (!this.props.isPublishEventActive)
+        // Given snapshot has no publish activity (!snapshot.resource.publishDate) and is the current approved (snapshot['@id'] === currentApprovedSnapshotID)
+        // The approval process is not currently in progress (!isApprovalActive)
+        const renderPublishLinkBool = this.props.allowPublishButton && !this.props.isPublishEventActive &&
+            (!snapshot.resource || !snapshot.resource.publishDate) && snapshot['@id'] === currentApprovedSnapshotID && !isApprovalActive;
```

Comment:
I still find this difficult to read, even with comments. Please put this into a function that returns a boolean value, possibly with `if...else` statement(s).

Reply:
As a result of other changes, the number of conditions in this statement has been reduced.  I think it's more readable, so I've left it as a single assignment statement.